src/SWP/Bundle/ContentBundle/Doctrine/ORM/RelatedArticleRepository.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2018 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2018 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\ContentBundle\Doctrine\ORM;
+
+use Doctrine\ORM\QueryBuilder;
+use Knp\Component\Pager\Pagination\PaginationInterface;
+use SWP\Bundle\ContentBundle\Doctrine\RelatedArticleRepositoryInterface;
+use SWP\Bundle\StorageBundle\Doctrine\ORM\EntityRepository;
+use SWP\Component\Common\Criteria\Criteria;
+use SWP\Component\Common\Pagination\PaginationData;
+
+class RelatedArticleRepository extends EntityRepository implements RelatedArticleRepositoryInterface
+{
+    public function getByCriteria(Criteria $criteria, array $sorting): array
+    {
+        $queryBuilder = $this->createQueryBuilder('ra');
+
+        $this->applyCustomCriteria($queryBuilder, $criteria);
+        $this->applyCriteria($queryBuilder, $criteria, 'ra');
```

Comment:
it can be replaced with 
`getQueryByCriteria` from trait.

Reply:
it can't be because I won't have access to QueryBuilder to which I want to apply custom criteria and I can't add more criteria to query builder after limiting is set.

Reply:
ahh... it returns query not query builder.